src/domain/core/src/services/pull_service.rs

```diff
@@ -132,6 +136,9 @@ pub struct PullMultiOptions {
     pub ingest_options: PollingIngestOptions,
     /// Sync-specific options,
     pub sync_options: SyncOptions,
+    /// Run compacting of all derivative datasets without saving data
+    /// if transformation fails
+    pub reset_derivatives: bool,
```

Comment:
Better to include in the parameter name some of the logic it carries -- we're not just doing `reset_derivatives`, but also running compacting

I don't know the best option to suggest right off the bat, but how about `run_save_compacting` (and just in the comment we will point out that save is not saving data, in case of failure)